utils/pytorch.py

```diff
@@ -85,16 +89,24 @@ def run(self, task_size: int, *args, **kwargs):
         """
 
         def runner_func():
-            with torch.cpu.amp.autocast() if self._do_autocast else nullcontext():
+            if self._do_autocast:
+                context = torch.cpu.amp.autocast()
+            elif self._gpu_autocast:
+                context = torch.cuda.amp.autocast()
+            else:
+                context = nullcontext()
+
+            with context:
                 start = time.time()
                 output = model(*args, **kwargs)
+                if self._gpu:
```

Comment:
maybe register finish time before this check and then overwrite previous value if self._gpu is True (to not include the if evaluation in cpu benchmark)